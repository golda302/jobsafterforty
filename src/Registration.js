import React from 'react';
import Dialog from 'material-ui/Dialog';
import FlatButton from 'material-ui/FlatButton';
import RaisedButton from 'material-ui/RaisedButton';
import TextField from 'material-ui/TextField';

export default class DialogRegistration extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      open: false,
    };
  }

  handleOpen = () => {
    this.setState({open: true});
  };

  handleClose = () => {
    this.setState({open: false});
  };
  
  render() {
    const style= {fontSize:12,display:'block'}
    const actions = [
      <FlatButton
        label="אישור"
        primary={true}
        keyboardFocused={true}
        onTouchTap={this.handleClose}
      />,
    ];

    return (
      <div>
      <div>
        <RaisedButton style={style} label="הרשמה" onTouchTap={this.handleOpen} />
        <Dialog
          title="הרשמה"
          actions={actions}
          modal={false}
          open={this.state.open}
          actionsContainerStyle={{textAlign:"left"}}
          style={{direction:'rtl',textAlign:'right'}}
          onRequestClose={this.handleClose}
        >
          <label class=""></label>
          <TextField hintText="נא למלא שדה זה" />
        </Dialog>
      </div>
      </div>
    );
  }
}