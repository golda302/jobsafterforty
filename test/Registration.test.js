import React from 'react';
import Registration from '../src/Registration.js';
import ShallowRenderer from 'react-test-renderer/shallow';
import Renderer from 'react-test-renderer';
import MuiThemeProvider from 'material-ui/styles/MuiThemeProvider';
import getMuiTheme from 'material-ui/styles/getMuiTheme';
import darkBaseTheme from 'material-ui/styles/baseThemes/darkBaseTheme';
  // Needed for onTouchTap
  import injectTapEventPlugin from 'react-tap-event-plugin';
  injectTapEventPlugin();
it('renders shallow correctly', () => {
  const renderer = new ShallowRenderer();
  renderer.render(<Registration/>);
  const tree = renderer.getRenderOutput();
  expect(tree).toMatchSnapshot();
});
it('renders correctly', () => {

  const tree = Renderer
    .create(<MuiThemeProvider muiTheme={getMuiTheme(darkBaseTheme)}><Registration/></MuiThemeProvider>)
    .toJSON();
  expect(tree).toMatchSnapshot();
});