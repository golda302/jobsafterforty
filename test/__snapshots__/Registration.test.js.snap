exports[`test renders correctly 1`] = `
<div>
  <div>
    <div
      className={undefined}
      style={
        Object {
          "WebkitTapHighlightColor": "rgba(0,0,0,0)",
          "backgroundColor": "#303030",
          "borderRadius": 2,
          "boxShadow": "0 1px 6px rgba(0, 0, 0, 0.12),
                 0 1px 4px rgba(0, 0, 0, 0.12)",
          "boxSizing": "border-box",
          "color": "rgba(255, 255, 255, 1)",
          "display": "inline-block",
          "fontFamily": "Roboto, sans-serif",
          "margin": 5,
          "minWidth": 88,
          "muiPrepared": true,
          "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
        }
      }>
      <button
        disabled={false}
        href={undefined}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchStart={[Function]}
        onTouchTap={[Function]}
        style={
          Object {
            "WebkitTapHighlightColor": "rgba(0, 0, 0, 0)",
            "backgroundColor": "#303030",
            "border": 10,
            "borderRadius": 2,
            "boxSizing": "border-box",
            "cursor": "pointer",
            "display": "inline-block",
            "fontFamily": "Roboto, sans-serif",
            "fontSize": "inherit",
            "fontWeight": "inherit",
            "height": 36,
            "lineHeight": "36px",
            "margin": 0,
            "muiPrepared": true,
            "outline": "none",
            "padding": 0,
            "position": "relative",
            "textAlign": "center",
            "textDecoration": "none",
            "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
            "verticalAlign": null,
            "width": "100%",
            "zIndex": 1,
          }
        }
        tabIndex={0}
        type="button">
        <div
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchStart={[Function]}>
          <div
            style={
              Object {
                "backgroundColor": false,
                "borderRadius": 2,
                "height": 36,
                "muiPrepared": true,
                "top": 0,
                "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
              }
            }>
            <span
              style={
                Object {
                  "color": "rgba(255, 255, 255, 1)",
                  "fontSize": 14,
                  "fontWeight": 500,
                  "letterSpacing": 0,
                  "margin": 0,
                  "muiPrepared": true,
                  "opacity": 1,
                  "paddingLeft": 16,
                  "paddingRight": 16,
                  "position": "relative",
                  "textTransform": "uppercase",
                  "userSelect": "none",
                }
              }>
              הרשמה
            </span>
          </div>
        </div>
      </button>
    </div>
  </div>
</div>
`;

exports[`test renders shallow correctly 1`] = `
<div>
  <div>
    <RaisedButton
      disabled={false}
      fullWidth={false}
      label="הרשמה"
      labelPosition="after"
      onTouchTap={[Function]}
      primary={false}
      secondary={false}
      style={
        Object {
          "margin": 5,
        }
      } />
    <Dialog
      actions={
        Array [
          <FlatButton
            disabled={true}
            fullWidth={false}
            keyboardFocused={true}
            label="אישור"
            labelPosition="after"
            labelStyle={Object {}}
            onKeyboardFocus={[Function]}
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
            onTouchStart={[Function]}
            onTouchTap={[Function]}
            primary={true}
            secondary={false} />,
        ]
      }
      actionsContainerStyle={
        Object {
          "textAlign": "left",
        }
      }
      autoDetectWindowHeight={true}
      autoScrollBodyContent={false}
      modal={false}
      onRequestClose={[Function]}
      open={false}
      repositionOnUpdate={true}
      style={
        Object {
          "direction": "rtl",
          "textAlign": "right",
        }
      }
      title="הרשמה">
      <label
        htmlFor="username">
        <font
          className="reqLabel">
          *
        </font>
        שם משתמש
      </label>
      <br />
      <TextField
        defaultValue=""
        disabled={false}
        floatingLabelFixed={false}
        fullWidth={false}
        hintText="נא למלא שדה זה"
        id="username"
        multiLine={false}
        name="username"
        onChange={[Function]}
        rows={1}
        type="text"
        underlineShow={true} />
      <br />
      <label
        htmlFor="email">
        <font
          className="reqLabel">
          *
        </font>
        כתובת אימייל
      </label>
      <br />
      <TextField
        defaultValue=""
        disabled={false}
        floatingLabelFixed={false}
        fullWidth={false}
        hintText="נא למלא שדה זה"
        id="email"
        multiLine={false}
        name="email"
        onChange={[Function]}
        rows={1}
        type="text"
        underlineShow={true} />
      <br />
      <label
        htmlFor="userpwd">
        <font
          className="reqLabel">
          *
        </font>
        סיסמה
      </label>
      <br />
      <TextField
        defaultValue=""
        disabled={false}
        floatingLabelFixed={false}
        fullWidth={false}
        hintText="נא למלא שדה זה"
        id="userpwd"
        multiLine={false}
        name="userpwd"
        onChange={[Function]}
        rows={1}
        type="password"
        underlineShow={true} />
      <br />
      <label
        htmlFor="userpwd">
        <font
          className="reqLabel">
          *
        </font>
        וידוא סיסמה
      </label>
      <br />
      <TextField
        defaultValue=""
        disabled={false}
        floatingLabelFixed={false}
        fullWidth={false}
        hintText="נא למלא שדה זה"
        id="userpwd2"
        multiLine={false}
        name="userpwd2"
        onChange={[Function]}
        rows={1}
        type="password"
        underlineShow={true} />
      <br />
      <Unknown
        formErrors={Object {}} />
    </Dialog>
  </div>
</div>
`;
